lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -523,7 +523,7 @@ public function getOrCreateDocuments($className, $nodes, array &$hints = array()
                     }
 
                     $targetDocument = isset($mapping['targetDocument']) ? $mapping['targetDocument'] : null;
-                    $coll = new ReferenceManyCollection($this->dm, $document, $mapping['property'], $referencedNodes, $targetDocument, $locale);
+                    $coll = new ReferenceManyCollection($this->dm, $document, $mapping['property'], $referencedNodes, $targetDocument, $locale, $this->getReferenceManyCollectionTypeFromMetadata($mapping));
```

Comment:
please change this to a multiline constructor call, like the one you add to below around line 1940.